<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#교차분석-과제" data-toc-modified-id="교차분석-과제-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>교차분석 과제</a></span><ul class="toc-item"><li><span><a href="#titanic데이터에서-sex와-class-간의-동질성-검정을-진행하세요." data-toc-modified-id="titanic데이터에서-sex와-class-간의-동질성-검정을-진행하세요.-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>titanic데이터에서 sex와 class 간의 동질성 검정을 진행하세요.</a></span></li></ul></li></ul></div>

#### 교차분석 
##### > 범주형 척도로 구성된 자료 간의 연관관계를 확인하기 위해 교차표를 만들어 관계를 확인하는 방법
##### > 각 범주에 따른 결과변수의 분포 설명
##### > 카이제곱 통계량 사용 : (관찰빈도-기대빈도)의 제곱의 합 / 기대빈도
##### > 기대빈도 : 두 변수가 독립일 때 이론적으로 기대할 수 있는 빈도 분포

#### 1. 적합성 검정 
##### > 각 범주에 따르는 데이터의 관측빈도 분포가 이론적으로 기대빈도 분포를 따르는지 검정
##### > 모집단 분포에 대한 가설이 옳게 됬는지 관측치와 비교 검정
##### > 관측치들이 어떠한 이론적 분포를 따르는지 관찰 가능
##### > 자유도 : 범주형 변수 수 - 1 ex) 산, 바다의 선호도 > 2 -1 = 1
#### 2. 독립성 검정
##### > 모집단이 두 개의 변수 A,B에 의해 범주화 되었을 때, 각 범주가 독립인지 검정
##### > 자우도 : (i-1) * (j-1) ex) 지역에 따른 구매한다/안한다? > (2-1) * (2-1) = 1

#### 3. 동질성 검정
##### > 범주 A에 대해 범주 B의 유형들이 같은 빈도로 나타나는지 검정
##### > 독립성 검정이랑 동일하게 시행 하지만 관점의 차이


# 교차분석 과제 
## titanic데이터에서 sex와 class 간의 동질성 검정을 진행하세요. 

In [2]:
import pandas as pd 

In [3]:
data = pd.read_csv('data/titanic.csv')
data

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True  

[891 rows x 15 columns]

#### 1. 가설설정
##### > H0 : sex는 class와 상관없이 동일하다.
##### > H1 : sex는 class에 따라 다르다.
#### 2. 유의수준 설정 : 0.05
#### 3. 기각역 설정 : 양측검정
#### 4. 전처리 및 통계량 계산(카이제곱 통계량)
##### > 범주 A에 대해 범주 B의 유형들이 같은 빈도로 분포되었는지 검정하기 위해, 카이제곱 통계량을 사용
#### 5. 해석 및 의사결정

In [5]:
#4. 전처리 및 통계량 계산(카이제곱 통계량)
data.info() # sex, class는 결측치 X

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   survived     891 non-null    int64  
 1   pclass       891 non-null    int64  
 2   sex          891 non-null    object 
 3   age          714 non-null    float64
 4   sibsp        891 non-null    int64  
 5   parch        891 non-null    int64  
 6   fare         891 non-null    float64
 7   embarked     889 non-null    object 
 8   class        891 non-null    object 
 9   who          891 non-null    object 
 10  adult_male   891 non-null    bool   
 11  deck         203 non-null    object 
 12  embark_town  889 non-null    object 
 13  alive        891 non-null    object 
 14  alone        891 non-null    bool   
dtypes: bool(2), float64(2), int64(4), object(7)
memory usage: 92.4+ KB


In [8]:
# 교차표 생성
table = pd.crosstab(data['sex'], data['class'])
table

class   First  Second  Third
sex                         
female     94      76    144
male      122     108    347

In [13]:
# 통계량 계산(카이제곱 통계량)
from scipy.stats import chi2_contingency
chi2, p, dof, expected = chi2_contingency(table) # 카이제곱 통계량, 유의확률, 자유도, 기대빈도
result = "chi2 : {}, p-value : {}, df : {}".format(chi2, p, dof)
print(result)
print(expected)


chi2 : 16.971499095517114, p-value : 0.00020638864348233114, df : 2
[[ 76.12121212  64.84399551 173.03479237]
 [139.87878788 119.15600449 317.96520763]]


##### 5. 해석 및 의사결정
###### > 카이제곱 통계량이 16.971 이며, 유의확률이 0.05 보다 작으므로 귀무가설을 기각하고 대립가설을 채택한다.
###### > 즉, class 관찰빈도는 sex 관찰빈도 수에 따라 다르게 나타난다.